src/com/goide/inspections/GoStringCannotBeNilInspection.java

```diff
@@ -0,0 +1,95 @@
+package com.goide.inspections;
+
+import com.goide.GoConstants;
+import com.goide.GoTypes;
+import com.goide.psi.*;
+import com.goide.psi.impl.GoElementFactory;
+import com.goide.psi.impl.GoTypeUtil;
+import com.intellij.codeInspection.LocalInspectionToolSession;
+import com.intellij.codeInspection.LocalQuickFix;
+import com.intellij.codeInspection.ProblemDescriptor;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiElement;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+public class GoStringCannotBeNilInspection extends GoInspectionBase{
+
+  private static final String DEFAULT_STRING = "\"\"";
+  private static final String PROBLEM_DESC = "String  cannot be nil";
+
+  @NotNull
+  @Override
+  protected GoVisitor buildGoVisitor(@NotNull ProblemsHolder holder, @NotNull LocalInspectionToolSession session) {
+    return new GoVisitor(){
+
+
+      @Override
+      public void visitVarSpec(@NotNull GoVarSpec o) {
+        super.visitVarSpec(o);
+        for(int i = 0; i < o.getExpressionList().size(); ++i){
+          check(o.getVarDefinitionList().get(i).getGoType(null), o.getVarDefinitionList().get(i).getValue());
+        }
+      }
+
+
+      @Override
+      public void visitAssignOp(@NotNull GoAssignOp o) {
+        super.visitAssignOp(o);
+        PsiElement[] right = o.getParent().getChildren();
+        if(right == null) return;
+
+        List<GoExpression> left = ((GoLeftHandExprList)right[0]).getExpressionList();
+        if(left == null) return;
+
+        for(int i = 0; i < left.size(); ++i){
+          GoExpression var = left.get(i);
+          PsiElement value = right[right.length - (left.size() - i)];
+
+          if(value instanceof GoExpression){
+            check(var.getGoType(null), (GoExpression)value);
+          }
+
+        }
+      }
+
+      @Override
+      public void visitBinaryExpr(@NotNull GoBinaryExpr o) {
+        super.visitBinaryExpr(o);
+
+        if(o.getOperator().getText().equals(GoTypes.EQ.toString()) ||
```

Comment:
`GoTypes.EQ.toString()` is just a debug string of element type, you should not use this in any logic.

Use `o.getOperator().getNode().getElementType() == GoTypes.EQ` instead